File: Strings/Java/Prefix_match_with_other_strings.java
class Solution
{
    public int klengthpref(String[] arr, int n, int k, String str) {
        str=str.substring(0,k);
        int count=0;
        for(int i=0; i<n; i++) {
            if(arr[i].length()<k) {
                continue;
            }
            String temp = arr[i].substring(0,k);
            if (temp.equals(str)) {
                count++;
            }
        }
        return count;
    }
}

File: Solutions/Python/Prefix_match_with_other_strings.py
class Solution:
    def klengthpref(self,arr,n,k,s):
        # return list of words(str) found in the board
        count = 0
        s1 = s[:k]
        for i in range(n):
            s2 = arr[i]
            if s1 == s2[:k]:
                count +=1    
        return count
